From dbb288d55a071a44eed1c9e21f5f835f416327be Mon Sep 17 00:00:00 2001
From: Zhenyu Wang <zhenyu.z.wang@intel.com>
Date: Thu, 31 Jul 2008 16:59:43 +0800
Subject: [PATCH] Fix SDVO reg definition

Remove wrong set tv resolution command, adding HDMI regs in dump.
(cherry picked from commit 76eb8e6f1f0c6962b23550564f4273f392567857)

diff --git a/src/i830_sdvo.c b/src/i830_sdvo.c
index d9b76d4..701aeaa 100644
--- a/src/i830_sdvo.c
+++ b/src/i830_sdvo.c
@@ -228,23 +228,30 @@ const static struct _sdvo_cmd_name {
     SDVO_CMD_NAME_ENTRY(SDVO_CMD_GET_SUPPORTED_POWER_STATES),
     SDVO_CMD_NAME_ENTRY(SDVO_CMD_GET_ENCODER_POWER_STATE),
     SDVO_CMD_NAME_ENTRY(SDVO_CMD_SET_ENCODER_POWER_STATE),
-    SDVO_CMD_NAME_ENTRY(SDVO_CMD_SET_TV_RESOLUTION_SUPPORT),
     SDVO_CMD_NAME_ENTRY(SDVO_CMD_SET_CONTROL_BUS_SWITCH),
     SDVO_CMD_NAME_ENTRY(SDVO_CMD_GET_SDTV_RESOLUTION_SUPPORT),
     SDVO_CMD_NAME_ENTRY(SDVO_CMD_GET_SUPPORTED_ENHANCEMENTS),
     /* HDMI op code */
     SDVO_CMD_NAME_ENTRY(SDVO_CMD_GET_SUPP_ENCODE),
     SDVO_CMD_NAME_ENTRY(SDVO_CMD_GET_ENCODE),
+    SDVO_CMD_NAME_ENTRY(SDVO_CMD_SET_ENCODE),
+    SDVO_CMD_NAME_ENTRY(SDVO_CMD_SET_PIXEL_REPLI),
     SDVO_CMD_NAME_ENTRY(SDVO_CMD_GET_PIXEL_REPLI),
     SDVO_CMD_NAME_ENTRY(SDVO_CMD_GET_COLORIMETRY_CAP),
+    SDVO_CMD_NAME_ENTRY(SDVO_CMD_SET_COLORIMETRY),
     SDVO_CMD_NAME_ENTRY(SDVO_CMD_GET_COLORIMETRY),
     SDVO_CMD_NAME_ENTRY(SDVO_CMD_GET_AUDIO_ENCRYPT_PREFER),
+    SDVO_CMD_NAME_ENTRY(SDVO_CMD_SET_AUDIO_STAT),
     SDVO_CMD_NAME_ENTRY(SDVO_CMD_GET_AUDIO_STAT),
     SDVO_CMD_NAME_ENTRY(SDVO_CMD_GET_HBUF_INDEX),
     SDVO_CMD_NAME_ENTRY(SDVO_CMD_SET_HBUF_INDEX),
     SDVO_CMD_NAME_ENTRY(SDVO_CMD_GET_HBUF_INFO),
     SDVO_CMD_NAME_ENTRY(SDVO_CMD_GET_HBUF_AV_SPLIT),
+    SDVO_CMD_NAME_ENTRY(SDVO_CMD_SET_HBUF_AV_SPLIT),
     SDVO_CMD_NAME_ENTRY(SDVO_CMD_GET_HBUF_TXRATE),
+    SDVO_CMD_NAME_ENTRY(SDVO_CMD_SET_HBUF_TXRATE),
+    SDVO_CMD_NAME_ENTRY(SDVO_CMD_SET_HBUF_DATA),
+    SDVO_CMD_NAME_ENTRY(SDVO_CMD_GET_HBUF_DATA),
 };
 
 static I2CSlaveAddr slaveAddr;
diff --git a/src/i830_sdvo_regs.h b/src/i830_sdvo_regs.h
index 747f2cd..4d55555 100644
--- a/src/i830_sdvo_regs.h
+++ b/src/i830_sdvo_regs.h
@@ -503,8 +503,6 @@ struct i830_sdvo_enhancements_arg {
 # define SDVO_DITHER_ON						(1 << 0)
 # define SDVO_DITHER_DEFAULT_ON					(1 << 1)
 
-#define SDVO_CMD_SET_TV_RESOLUTION_SUPPORT		0x93
-
 #define SDVO_CMD_SET_CONTROL_BUS_SWITCH			0x7a
 # define SDVO_CONTROL_BUS_PROM				(1 << 0)
 # define SDVO_CONTROL_BUS_DDC1				(1 << 1)
-- 
1.5.6.4

